### DISCUSSION TITLE
Is it possible to run some code in the middle of text of the md files?

### DISCUSSION THREAD

**Discussion by gsuarezr** [alshedivat/al-folio#2121] [Q&A] [answered]
For example let's say than in the about page I would like to have a counter for how many days have passed this year, is it possible to do this from the about.md file or should I take that text to the about.liquid file and do it there?

**Comment by george-gca** [comment]
It could be on any of them, depending on how and where you want to use/display it.

**Reply by gsuarezr** [reply]
Could you point to a minimal example where something like that is done on the about.md file? or give some direction. I know how to do it in the about.liquid file, but I'm not sure how to use code in the .md file. 

Thanks a lot for the quick reply

**Reply by george-gca** [reply]
The md file is a markdown in [GitHub Flavored Markdown](https://github.github.com/gfm/) as set in [_config.yml](https://github.com/alshedivat/al-folio/blob/1d84621f220e157d0701d5cbd69f334cb6730a4c/_config.yml#L196), so it accepts html code as well, as you can see in [_pages/blog.md](https://raw.githubusercontent.com/alshedivat/al-folio/master/_pages/blog.md). Any html code you would use in `about.liquid` could also be used in `about.md`.

If you want to use the information in the about text, for example, you should put it in `about.md`. If it is in some other way, probably you'll want to put it in `about.liquid`.

Note that all liquid code is processed at build time. What this means is, if you try to do this date diff using liquid it would only be calculated during build time, so unless you build your site everyday the value generated using the liquid code would not be updated. You must do this date diff using javascript.

As for how to do this, I never did it myself so you'll probably need to do some Googling. The term you are looking for is `javascript days since date`. [This looks like](https://www.geeksforgeeks.org/how-to-calculate-the-number-of-days-between-two-dates-in-javascript/) a good solution for example.

**Reply by gsuarezr** [reply]
Sure I understand what you mean, the example was just meant to be anything that couldn't be harcoded, the part that confuses me a bit is how to output the result of javascript code inside the md file, I kind of wanted to know how its possible to run javascript here, which I know understand is possible.    is it as simple as using <script> tags inside the md file ?

**Reply by george-gca** [reply, marked as answer]
Yes. The md file supports regular html code.

**Reply by george-gca** [reply]
@mcditoos just found [this link](https://github.com/github/relative-time-element). Check if it is useful for you.

**Reply by gsuarezr** [reply]
Thanks a lot George. Indeed the link was useful. Amazing to have such nice, attentive support on an open source project
